library/core/src/time.rs

```diff
@@ -508,6 +534,7 @@ impl Duration {
     #[rustc_const_stable(feature = "duration_consts", since = "1.32.0")]
     #[must_use]
     #[inline]
+    #[ensures(|ms| *ms == self.nanos.0 / NANOS_PER_MILLI)]
```

Comment:
With the above comments (or perhaps preconditions) you'd want to do the same for `NANOS_PER_MILLI`.